docs/index.md

```diff
@@ -116,51 +116,60 @@ $ npm test
 
 ## Run Cycle Overview
 
-A brief outline on the order Mocha's components are executed.
-Worth noting that all hooks, `describe` and `it` callbacks are run in the order they are defined (i.e. found in the file).
-
-```
-run 'mocha spec.js'
-|
-spawn child process
-|
-|--------------> inside child process
-  process and apply options
-  |
-  run spec file/s
-  |
-  |--------------> per spec file
-    suite callbacks (e.g., 'describe')
-    |
-    'before' root-level pre-hook
-    |
-    'before' pre-hook
-    |
-    |--------------> per test
-      'beforeEach' root-level pre-hook
-      |
-      'beforeEach' pre-hook
-      |
-      test callbacks (e.g., 'it')
-      |
-      'afterEach' post-hook
-      |
-      'afterEach' root-level post-hook
-    |<-------------- per test end
-    |
-    'after' post-hook
-    |
-    'after' root-level post-hooks
-  |<-------------- per spec file end
-|<-------------- inside child process end
-```
+> Updated for v9.0.0.
+
+The following is a mid-level outline of Mocha's "flow of execution" when run in Node.js; the "less important" details have been omitted.
+
+In a browser, test files are loaded by `<script>` tags, and calling `mocha.run()` begins at step 9 [below](#serial-mode).
+
+### Serial Mode
+
+1. User (that's you) executes `mocha`
+2. Loads options from config files, if present
+3. Mocha processes any command-line options provided
+4. If known flags for the `node` executable are found:
+   1. Mocha will spawn `node` in a child process, executing itself with these flags
+   2. Otherwise, Mocha does not spawn a child process
+5. Mocha loads modules specified by `--require`
+   1. If a file loaded this way contains known Mocha-specific exports (e.g., [root hook plugins]), Mocha "registers" these
+   2. If not, Mocha ignores any exports of a `--require`'d module
+6. Mocha _discovers_ test files; when given no files or directories, it finds files with extensions `.js`, `.mjs` or `.cjs` in the `test` directory (but not its children), relative to the current working directory
```

Comment:
Do we validate `--reporter` and `--ui` a step before this?

Reply:
yes